ts/editor/NoteEditor.svelte

```diff
@@ -435,223 +448,230 @@ the AddCards dialog) should be implemented in the user of this component.
-                                        richTextsHidden[index] = true;
-                                    }
-                                }}
-                                --icon-align="bottom"
-                            >
-                                <svelte:fragment slot="field-name">
-                                    <LabelName>
-                                        {field.name}
-                                    </LabelName>
-                                </svelte:fragment>
-                                <FieldState>
-                                    {#if cols[index] === "dupe"}
-                                        <DuplicateLink />
-                                    {/if}
-                                    {#if plainTextDefaults[index]}
-                                        <RichTextBadge
-                                            show={!fieldsCollapsed[index] &&
-                                                (fields[index] === $hoveredField ||
-                                                    fields[index] === $focusedField)}
-                                            bind:off={richTextsHidden[index]}
-                                            on:toggle={async () => {
-                                                richTextsHidden[index] =
-                                                    !richTextsHidden[index];
-
-                                                if (!richTextsHidden[index]) {
-                                                    refocusInput(
-                                                        richTextInputs[index].api,
-                                                    );
-                                                }
-                                            }}
-                                        />
-                                    {:else}
-                                        <PlainTextBadge
-                                            show={!fieldsCollapsed[index] &&
-                                                (fields[index] === $hoveredField ||
-                                                    fields[index] === $focusedField)}
-                                            bind:off={plainTextsHidden[index]}
-                                            on:toggle={async () => {
-                                                plainTextsHidden[index] =
-                                                    !plainTextsHidden[index];
-
-                                                if (!plainTextsHidden[index]) {
-                                                    refocusInput(
-                                                        plainTextInputs[index].api,
-                                                    );
-                                                }
-                                            }}
+    {#if isShowMaskEditor}
+        <MaskEditor path={maskEditorImagePath} />
+    {:else}
```

Comment:
I have used if else block to show and hide the mask editor and note editor. But this implementation have issues. When switching between the two editor the shapes drawn data removed. I have tried to fix but could not fixed.

And second issues is
```
Uncaught TypeError: Cannot read properties of null (reading 'move')
    at expandTags (note_creator.js:134023:20)
    at snapResizer (note_creator.js:134032:9)
    at note_creator.js:134026:45
```

I think the implementation should be changed.

Reply:
Perhaps you could avoid that issue by toggling css visibility instead?